lib/middleware/fires_middleware.dart

```diff
@@ -25,14 +26,100 @@ Middleware<AppState> _createLoadFires() {
       String url = endpoints['getFires'];
       final response = await http.get(url);
       final responseData = json.decode(response.body)['data'];
-      List fires = responseData.map((model) => Fire.fromJson(model)).toList();
+      List<Fire> fires = responseData.map<Fire>((model) => Fire.fromJson(model))
+          .toList();
+      print("load fires");
+      fires = _calculateFireImportance(fires);
       store.dispatch(new FiresLoadedAction(fires));
     } catch (e) {
+      print(e);
+      print(e.stackTrace);
       store.dispatch(new FiresLoadedAction([]));
     }
   };
 }
 
+/// Calculates the importance of each fire
+
+const numberOfFires = "numberOfFires";
+const topImportance = "topImportance";
+const average = "average";
+
+List<Fire> _calculateFireImportance(List<Fire> fires) {
+  var firesStatus = {
+    numberOfFires: 0,
+    topImportance: 0.0,
+    average: 0.0
+  };
+  for (var fire in fires) {
+    fire.importance = _calculateImportanceValue(fire, firesStatus);
+  }
+  for (var fire in fires) {
+    fire.scale = _getPonderatedImportanceFactor(fire.importance, fire.statusCode, firesStatus) / 1.5;
+  }
+
+  return fires;
+}
+
+
+double _calculateImportanceValue(Fire data, Map<String, dynamic> status) {
+  var manFactor = 1.0;
+  var terrainFactor = 3.0;
+  var aerialFactor = 7.0;
+
+  var importance = data.human * manFactor + data.terrain * terrainFactor +
+      data.aerial * aerialFactor;
+
+  status[numberOfFires] += 1;
+  if (status[topImportance] < importance) {
+    status[topImportance] = importance;
```

Comment:
this looks good, but i feel like these functions should not be mixed with middleware stuff. we should try to keep things tidy and clean. maybe create a separate file with these?

Reply:
created an `utils` folder